app/Http/Controllers/RecipeController.php

```diff
@@ -60,281 +62,177 @@ public function create()
-   */
-    public function favorite(Recipe $recipe, Request $request)
+    /**
+     *Makes a recipe favorite to a user
+     */
+    public function favorite(Recipe $recipe)
     {
-        try {
-            $this->authorize('update', $recipe);
-            $recipe->is_favorite = !$recipe->is_favorite;
-            $recipe->save();
-            return redirect()->back();
-        } catch (Exception $e) {
-            session()->flash('alert', [
-                'title' => 'Error!',
-                'message' => $e->getMessage(),
-                'type' => 'error'
-            ]);
-            return redirect()->back();
-        }
+        $this->authorize('update', $recipe);
+        $recipe->is_favorite = !$recipe->is_favorite;
+        $recipe->save();
+
+        return redirect()->back();
     }
 
-    /*
-        Creates a review(rate the recipe)
-       */
+    /**
+     * Creates a review(rate the recipe)
+     */
     public function rate(Recipe $recipe, Request $request)
     {
-        try {
-            $this->authorize('view', $recipe);
-
-            $rating = $recipe->reviews()->where("user_id", Auth::user()->id)->first();
-
-            $request->request = $request->validate([
-                "rating" => ['required', 'integer', 'max:5', 'min:1'],
-                "msg" => ['required', 'string', 'max:500', 'min:1']
-            ],
-                [
-                    "rating.required" => "Rating is required!",
-                    "msg.required" => "Message is required!",
-                ]);
-            if ($rating) {
-                $rating->rating = $request->rating;
-                $rating->message = $request->msg;
-                $rating->save();
-            }
-            if ($rating === null) {
-                $rating = Review::create([
-                    "rating" => $request->rating,
-                    "message" => $request->msg,
-                    "user_id" => Auth::user()->id,
-                    "recipe_id" => $recipe->id
-                ]);
-            }
-            NotificationF::send(User::find($recipe->user_id), new RecipeRated($recipe->title,$rating->rating,Auth::user()));
-            session()->flash('alert', [
-                'title' => 'Success!',
-                'message' => 'Review added successfully.',
-                'type' => 'success'
-            ]);
-
-            return redirect()->back();
-        } catch (Exception $e) {
-            session()->flash('alert', [
-                'title' => 'Error!',
-                'message' => $e->getMessage(),
-                'type' => 'error'
+        $this->authorize('view', $recipe);
+        $request->request = $request->validate([
+            "rating" => ['required', 'integer', 'max:5', 'min:1'],
+            "msg" => ['required', 'string', 'max:500', 'min:1']
+        ],
+            [
+                "rating.required" => "Rating is required!",
+                "msg.required" => "Message is required!",
```

Comment:
Validation should be done on front-end , custom messages from back are not option here